assets/js/blocks/classic-template/archive-product.ts

```diff
@@ -0,0 +1,59 @@
+/**
+ * External dependencies
+ */
+import {
+	getBlockType,
+	createBlock,
+	createBlocksFromInnerBlocksTemplate,
+	type BlockInstance,
+} from '@wordpress/blocks';
+
+/**
+ * Internal dependencies
+ */
+import {
+	INNER_BLOCKS_TEMPLATE as productsInnerBlocksTemplate,
+	QUERY_DEFAULT_ATTRIBUTES as productsQueryDefaultAttributes,
+} from '../product-query/constants';
+import { VARIATION_NAME as productsVariationName } from '../product-query/variations/product-query';
+
+const createRowBlock = ( innerBlocks: Array< BlockInstance > ) => {
+	const rowVariationName = `group-row`;
+	const groupBlockVariations = getBlockType( 'core/group' )?.variations || [];
+	const rowVariation = groupBlockVariations.find(
+		( { name }: { name: string } ) => name === rowVariationName
+	);
+
+	if ( ! rowVariation ) return null;
```

Comment:
Nit: we tend to always use multiline clauses:

```suggestion
	if ( ! rowVariation ) {
		return null;
	}
```